packages/run-protocol/src/runStake/runStakeKit.js

```diff
@@ -212,17 +213,17 @@ export const makeRunStakeKit = (zcf, startSeat, manager) => {
     const debt = getCurrentDebt();
     const collateral = getCollateralAllocated(vaultSeat);
 
-    const giveColl = proposal.give[KW.Attestation] || emptyCollateral;
-    const wantColl = proposal.want[KW.Attestation] || emptyCollateral;
+    const giveColl = proposal.give.Attestation || emptyCollateral;
+    const wantColl = proposal.want.Attestation || emptyCollateral;
 
     // new = after the transaction gets applied
     const newCollateral = addSubtract(collateral, giveColl, wantColl);
     // max debt supported by current Collateral as modified by proposal
     const { amountLiened, maxDebt: newMaxDebt } =
       manager.maxDebtForLien(newCollateral);
 
-    const giveRUN = AmountMath.min(proposal.give[KW.Debt] || emptyDebt, debt);
-    const wantRUN = proposal.want[KW.Debt] || emptyDebt;
+    const giveRUN = AmountMath.min(proposal.give.Debt || emptyDebt, debt);
```

Comment:
LGTM.

> proposal.give.Attestation looks nicer, but it's a use that is not checked against any defining occurrence.

So now someone could type `proposal.give.Debt` and get no error, whereas `proposal.give[KW.Debte]` would have said `KW` doesn't have that key. But in both cases `give` doesn't constrain the keys.

I think where we want to get to is that you can validate the proposal with pattern matching so that the type system has a guarantee from some runtime validation as to what's in `give`. So imo let's work towards that and not invest in the stop-gap.